### PR TITLE
Add new parameters for types

### PR DESCRIPTION
**Pull request by phanpak**
related to https://github.com/brinktrade/brink-sdk/pull/100#pullrequestreview-1636050737

### CODE REVIEW

File: src/Types.ts [hunk 1]
```diff
@@ -164,8 +165,8 @@ export interface ConditionArgsBase {
 
 export interface PriceConditionArgs extends ConditionArgsBase {
   operator: PriceOperator
-  tokenA: string
```
Comment (mikec) [inline src/Types.ts:167]: @phanpak I think these should actually be `tokenA: string | TokenArgs` and the user can provide either string for address, or full TokenArgs object

Comment (mikec) [reply src/Types.ts:167]: I did this for market swap action, but I guess I forgot to do it for price condition:
https://github.com/brinktrade/brink-types/blob/30df9d78c861a4126a64bed5afb092f63fbdde9c/src/Types.ts#L196

Comment (mikec) [reply src/Types.ts:167]: btw the TokenArgs type with additional params is really for future NFT support, I think we'll be encouraging devs to set address here in the near future. But good to allow TokenArgs now I think

File: src/Types.ts [hunk 2]
```diff
@@ -706,3 +709,10 @@ export interface SubmitStrategyRequest {
 export interface SubmitStrategyResponse {
   hash: string
 }
+
+export interface requireBitUsedRequest {
+    bitmapIndex: BigIntish;
+    bit: BigIntish;
```
Comment (phanpak) [inline src/Types.ts:715]: can someone help me determine if it's all the required params? what i did essentially was to check the arguments of the function here: https://github.com/brinktrade/brink-strategies/blob/master/src/Primitives/Primitives01.sol#L71C12-L71C26

File: src/Types.ts [hunk 3]
```diff
@@ -706,3 +709,15 @@ export interface SubmitStrategyRequest {
 export interface SubmitStrategyResponse {
   hash: string
 }
+
+export interface requireBitUsedRequest {
+  bit: BitJSON;
```
Comment (phanpak) [inline src/Types.ts:714]: @mikec or @MatiasOS when defining a primitiveArgs. should it nest this inside a `bit` key?

File: src/Types.ts [hunk 4]
```diff
@@ -125,8 +125,11 @@ export type PrimitiveFunctionName =
   'requireBlockNotMined' |
   'requireBlockMined' |
   'requireUint256LowerBound' |
+  'requireUint256UpperBound' |
   'limitSwapExactInput' |
-  'blockInterval'
+  'blockInterval' | 
+  'requireBitNotUsed' |
```
Comment (mikec) [inline src/Types.ts:131]: I added these so you'll have to rebase and resolve the conflict

File: src/Types.ts [hunk 5]
```diff
@@ -539,6 +542,14 @@ export interface RequireUint256LowerBoundRequest extends RequireCheckRequest {
 export interface RequireUint256LowerBoundResponse extends RequireCheckResponse {
   oracleValue: string
 }
+export interface RequireUint256UpperBoundRequest extends RequireCheckRequest {
```
Comment (mikec) [inline src/Types.ts:545]: these request and response classes were intended to be used in API, but I don't think API is using them. I think we should not add these for now. I think these will be necessary if we build a TypeScript API client, but that's not super high priority yet

File: src/Types.ts (hunk 3 above)
Comment (mikec) [reply src/Types.ts:722]: @phanpak I followed the pattern we're using for the UseBit primitive and used index/value. We're trying to keep all user facing naming to index/value. See https://github.com/brinktrade/brink-sdk/commit/65c1f2723a631cc6d36379034c638256bfc51a28#diff-02e99e848089f1a0a30aa7b26396bfd88f111a0d5914da4929901d3281e5cd98R21

For types repo, we don't need to add these request/response types here (see above)

**Review by mikec — CHANGES_REQUESTED** [review]

File: src/Types.ts (hunk 4 above)
Comment (mikec) [reply src/Types.ts:133]: @phanpak need to fix the conflict with base here still

Comment (phanpak) [reply src/Types.ts]: sorry i didn't get it. done!

**Review by mikec — APPROVED** [review]

**Pull request merged**